pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -247,13 +255,26 @@ type Workload struct {
 	// A Workload object.
 	Workload *unstructured.Unstructured
 
+	// Unready indicates whether this resource is unready to apply
+	// because of dependency not ready
+	Unready bool
+
 	// Traits associated with this workload.
-	Traits []unstructured.Unstructured
+	Traits []*Trait
 
 	// Scopes associated with this workload.
 	Scopes []unstructured.Unstructured
 }
 
+// A Trait produced by an OAM ApplicationConfiguration.
+type Trait struct {
+	Object unstructured.Unstructured
+
+	// Unready indicates whether this resource is unready to apply
+	// because of dependency not ready
+	Unready bool
```

Comment:
The reason is that the default of bool is false.
The default of any Workload/Trait is actually ready (= !unready). The dataInputs field might not exist so it might not go into the dependency logic.

What about change the name to `HasDep`?

Reply:
I have changed this to HasDep. I think having a default false value is still very important in preventing mistaken init place.